### ISSUE TITLE
linkAll with hasOne relationship not linking correctly

### ISSUE THREAD

**Issue by Patrik-Lundqvist** [js-data/js-data#148]
I'm having issues with linkAll not linking objects with a hasOne relationship. Let's say a user can optionally select an avatar to use from a list, this creates a hasOne relation between the user and the avatar.

Like:
```javascript
DS.defineResource({
    name: 'user',
    idAttribute: 'id',
    endpoint: 'users',
    relations: {
        hasOne: {
          avatar: {
              localKey: 'avatarId',
              localField: 'avatar'
          }
        }
    }
});
DS.defineResource({
    name: 'avatar',
    idAttribute: 'id',
    endpoint: 'avatars'
});
```

If I were to insert an avatar and then run linkAll on users, shouldn't their avatar links be updated?

See plunker: [http://plnkr.co/edit/8B2vB5](http://plnkr.co/edit/8B2vB5)

**Comment by jmdobry** [comment]
I fixed this, but I think the fix might be in the 2.0 branch.

**Comment by Patrik-Lundqvist** [comment]
You're right, seems to be working in the 2.0 release :) Thanks
